handsontable/src/stylesManager.js

```diff
@@ -3,31 +3,197 @@
  */
 export class StylesManager {
   /**
-   * The Handsontable instance.
+   * The Handsontable instance's root element.
+   *
+   * @type {HTMLElement}
+   */
+  #rootElement;
+
+  /**
+   * The computed style of the root element.
+   *
+   * @type {CSSStyleDeclaration}
+   * @private
    */
-  #hot;
+  #rootComputedStyle;
+
+  /**
+   * `true` if the legacy theme is enabled, `false` otherwise.
+   *
+   * @type {boolean}
+   */
+  static #legacyTheme = false;
+
   /**
    * Keeps a value of cell vertical padding css variable.
    *
    * @type {number}
    */
-  cellVerticalPadding;
+  static #cellVerticalPadding;
```

Comment:
Using static fields may break styling on the page where multiple table instances are created, unless I'm wrong? If that is ok I could suggest rewriting static class fields to basic variables defined above the class. I have the impression that static fields increase the size of the code making it less readable.

Reply:
It's probably not a common use case to have multiple instances with different themes on one page, but you're right, this won't work in that scenario. I'll change it 👍